README.md

```diff
@@ -8,14 +8,76 @@ There are multiple test suites, each of which is detailed below.
 
 ## Http Signatures Version 11 Tests
 
-This test suite will check any application that generates [Http Signaturesl](hhttps://tools.ietf.org/html/draft-cavage-http-signatures-11) Headers to
+This test suite will check any application that generates [Http Signatures](hhttps://tools.ietf.org/html/draft-cavage-http-signatures-11) Headers to
 ensure conformance with the specification.
 
+
+### Creating a Binary
+You web application will need to be accessible from the command line.
+It will also need to accept the following command line parameters:
+
+```
+Usage: http-signature-header [options] [command]
+
+Options:
+  -H, --headers <headers>                   A list of header names.
+  -k, --keyId <keyId>                       A Key Id <string>.
+  -pk, --private-key <privateKey>.          A private key file name <filename>.
+  -pkt, --private-key-type <privateKeyType  The type of the private key.
+  -a, --algorithm [algorithm]               One of: rsa-sha1, hmac-sha1, rsa-sha256, hmac-sha256, hs2019.
+  -c, --created [created]                   The created param for the signature.
+  -e, --expires [expires]                   The expires param for the signature.
+  -h, --help                                output usage information
+
+Commands:
+  c14n|C
+  sign|S [options]
+  verify|V
+```
+All tests will run against your binary and assume that an exit code greater
+than 0 represents an error.
+Your binary will receive an [http message](https://developer.mozilla.org/en-US/docs/Web/HTTP/Messages) via [standard in](https://en.wikipedia.org/wiki/Standard_streams):
+
+```
```

Comment:
You may want to say something to the effect of "This is an example of the type of input your program will receive in addition to the command line options."